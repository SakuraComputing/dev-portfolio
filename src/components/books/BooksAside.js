import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Book from '../../components/books/Book';

class BooksAside extends Component {
    render() {

        const { books } = this.props;

        let singleBook;

        if(books.length > 0) {
            singleBook = books.map(book => (
                <Book book={book}/>
            ));
        }

        return (
            <aside className="books aside section">
                <div className="section-inner">
                    <h2 className="heading">Latest Reading</h2>
                    <div className="content">
                        <ul className="list-unstyled">
                            {singleBook}
                        </ul>
                    </div>
                </div>
            </aside>
        );
    }
}

BooksAside.propTypes = {
    Books: PropTypes.array
};

export default BooksAside;
