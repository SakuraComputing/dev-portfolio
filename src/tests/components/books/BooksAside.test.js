import React from 'react';
import { shallow } from 'enzyme';
import BooksAside from '../../../components/books/BooksAside';

describe('Latest Book Section', () => {
    it('should render correctly', () => {
        const books = jest.fn();
        const wrapper = shallow(<BooksAside books={books} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render nothing if no books are passed in', ()=> {
        const books = [];
        const wrapper = shallow(<BooksAside books={books} />);
        expect(wrapper).toMatchSnapshot();
    });
});
