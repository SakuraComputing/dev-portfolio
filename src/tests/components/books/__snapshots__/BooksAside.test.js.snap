// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Latest Book Section should render correctly 1`] = `
<aside
  className="books aside section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Latest Reading
    </h2>
    <div
      className="content"
    >
      <ul
        className="list-unstyled"
      />
    </div>
  </div>
</aside>
`;

exports[`Latest Book Section should render nothing if no books are passed in 1`] = `
<aside
  className="books aside section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Latest Reading
    </h2>
    <div
      className="content"
    >
      <ul
        className="list-unstyled"
      />
    </div>
  </div>
</aside>
`;
